packages/editor-ui/src/components/CanvasChat/CanvasChat.vue

```diff
@@ -175,28 +175,18 @@ const closePanel = () => {
 // This function creates a promise that resolves when the workflow execution completes
 // It's used to handle the loading state while waiting for the workflow to finish
 async function createExecutionPromise() {
-	let resolvePromise: () => void;
-	const promise = new Promise<void>((resolve) => {
-		resolvePromise = resolve;
-	});
-
-	// Watch for changes in the workflow execution status
-	const stopWatch = watch(
-		() => workflowsStore.getWorkflowExecution?.status,
-		(newStatus) => {
-			// If the status is no longer 'running', resolve the promise
-			if (newStatus && newStatus !== 'running') {
-				resolvePromise();
-				// Stop the watcher when the promise is resolved
-				stopWatch();
+	return await new Promise<void>((resolve) => {
+		const resolveIfFinished = (isRunning: boolean) => {
```

Comment:
Stopping watcher in separate function due to possible `unwatch used before initialization` error, if it runs on the first attempt. This could happen if for any reason `runWorkflow` (the prerequisite for this promise) fails.